processor/sumologicprocessor/config.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sumologicprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/sumologicprocessor"
+
+import (
+	"go.opentelemetry.io/collector/component"
+)
+
+type Config struct {
+	AddCloudNamespace           bool                      `mapstructure:"add_cloud_namespace"`
+	TranslateAttributes         bool                      `mapstructure:"translate_attributes"`
+	TranslateTelegrafAttributes bool                      `mapstructure:"translate_telegraf_attributes"`
+	NestAttributes              *NestingProcessorConfig   `mapstructure:"nest_attributes"`
+	AggregateAttributes         []aggregationPair         `mapstructure:"aggregate_attributes"`
+	LogFieldsAttributes         *logFieldAttributesConfig `mapstructure:"field_attributes"`
+	TranslateDockerMetrics      bool                      `mapstructure:"translate_docker_metrics"`
+}
+
+type aggregationPair struct {
+	Attribute string   `mapstructure:"attribute"`
+	Prefixes  []string `mapstructure:"prefixes"`
+}
+
+const (
+	defaultAddCloudNamespace           = true
+	defaultTranslateAttributes         = true
+	defaultTranslateTelegrafAttributes = true
+	defaultTranlateDockerMetrics       = false
+
+	// Nesting processor default config
+	defaultNestingEnabled            = false
+	defaultNestingSeparator          = "."
+	defaultNestingSquashSingleValues = false
+
+	defaultAddSeverityNumberAttribute = false
+	defaultAddSeverityTextAttribute   = false
+	defaultAddSpanIDAttribute         = false
+	defaultAddTraceIDAttribute        = false
+)
+
+var (
+	defaultAggregateAttributes = []aggregationPair{}
+)
+
+// Ensure the Config struct satisfies the config.Processor interface.
+var (
+	_                     component.Config = (*Config)(nil)
+	defaultNestingInclude                  = []string{}
+	defaultNestingExclude                  = []string{}
```

Comment:
> make this a immutable

How to do it?

This is declared as var to be easily accessible and give some clearness about default values, and because golang does not allow to create constant with `[]string{}` type

Reply:
> How to do it?

Define a read only interface. Another way to get around the race detector would be to define these per configuration created which still provides context of the vars but doesn't allow them to be shared across configurations.